examples/list-streams.js

```diff
@@ -0,0 +1,188 @@
+    const facilityId = 'YOUR_FACILITY_URN';
+    const facility = await getFacility(token, facilityId);
+    const defaultModelId = facilityId.replace('urn:adsk.dtt:', 'urn:adsk.dtm:');
+    const defaultModel = facility.links.find((m) => {
+        return  m.modelId === defaultModelId;
+    });
+
+    // STEP 3 - get streams and print stream + its parent
+    const streams = await getStreams(token, defaultModel.modelId);
+
+    for (const stream of streams) {
+        // host is stored as parent
+        const parentXref = stream['x:p'];
+
+        if (!parentXref) {
+            continue;
+        }
+        // the id of the host is encoded
+        const [ modelId, key ] = decodeXref(parentXref);
+
+        const elementData = await getElementData(token, `urn:adsk.dtm:${modelId}`, key);
+
+        if (!elementData) {
+            continue;
+        }
+        // print out name of stream + name of parent
+        console.log(`${stream['n:n']}:${elementData['n:n']}`);
+    }
+}
+
+/**
+ * Creates 2-legged token using provided inputs.
+ * @param {string} clientID 
+ * @param {string} clientSecret 
+ * @param {string} scope 
+ * @returns {Promise<string>}
+ */
+async function createToken(clientID, clientSecret, scope) {
+    const auth = Buffer.from(`${clientID}:${clientSecret}`).toString('base64');
+    const options = new URLSearchParams({
+        'grant_type': 'client_credentials',
+        'scope': scope
+    });
+
+    const tokenResponse = await fetch(`https://developer.api.autodesk.com/authentication/v2/token?${options}`, {
+        method: 'POST',
+        headers: {
+            'Authorization': `Basic ${auth}`
+        }
+    });
+
+    const token = await tokenResponse.json();
+
+    return token.access_token;
+}
+
+/**
+ * Returns facility based on given URN.
+ * @param {string} token 
+ * @param {string} urn 
+ * @returns {Promise<object>}
+ */
+async function getFacility(token, urn) {
+    const response = await fetch(`https://tandem.autodesk.com/api/v1/twins/${urn}`, {
+        method: 'GET',
+        headers: {
+            'Authorization': `Bearer ${token}`
+        }
+    });
+
+    const data = await response.json();
+
+    return data;
+}
+
+/**
+ * Return array of stream elements from given model. The elements include standard (n) and xref (x) properties.
+ * @param {string} token 
+ * @param {string} urn 
+ * @returns {Promise<object[]>}
+ */
+async function getStreams(token, urn) {
+    const inputs = {
+        families: [ 'n', 'x'],
+        includeHistory: false,
+        skipArrays: true
+    };
+    const response = await fetch(`https://tandem.autodesk.com/api/v2/modeldata/${urn}/scan`, {
+        method: 'POST',
+        headers: {
+            'Authorization': `Bearer ${token}`
+        },
+        body: JSON.stringify(inputs)
+    });
+
+    const data = await response.json();
+    const results = [];
+
+    for (const item of data) {
+        if ((item['n:a'] & 0x01000003) === 0x01000003) {
```

Comment:
That would be reasonable to begin extracting these constants into a dedicated file. In the future we might want to ship some package like `tandem-core`